### ISSUE TITLE
Global search window issue 

### ISSUE THREAD

**Issue by LoayGhreeb** [JabRef/jabref#11024]
While working on PR #11010, I noticed an issue that needs to be solved first:

- When using the global search window, the search results always change according to the active search query of the active tab. This allows for updating the query without the need to close the window and search again. However, a problem arises when changing the tab while opening this window, as the search results will change according to the active query of this tab.

	https://github.com/JabRef/jabref/assets/52158423/11c89797-a135-4fae-8368-e53f3dc6cd9d

Possible Solutions:

- Stop changing the search result according to the active search query. This could become difficult if the query needs to be changed frequently, as it would require closing the window, writing the new query, and opening the window again.
- Copy the search query to all active tabs (which might not be a good solution) as a complex query made in another tab could be lost.
- Use a separate search bar in the global search window.

**Comment by koppor** [comment]
> * Use a separate search bar in the global search window.

Good idea. The initial value is the current search. Then, it works as of now.

> Copy the search query to all active tabs

Make this a button in the global search dialog

**Comment by LoayGhreeb** [comment]
@koppor, what about always enabling the `keepSearchString` toggle after opening the global search dialog?

**Comment by koppor** [comment]
> @koppor, what about always enabling the `keepSearchString` toggle after opening the global search dialog?

When closing the global search, it gets back to its original state. If the state is "off", the previous searches are restored? I did not try. I would assume "no". -- Therefore the proposal with the separate search string in the popup. This is also more natural, because the search results are close to the search query. I know that selecting an entry could be difficult, if the query of a table does not match that entry. In that case, I would just clear the search in the table and then select the entry?

**Comment by LoayGhreeb** [comment]
> if the query of a table does not match that entry. In that case, I would just clear the search in the table and then select the entry?

What should be done in this case? Should the tab be switched and let it be figured out that the table has been filtered, or should the current search just be cleared?

**Comment by koppor** [comment]
> > if the query of a table does not match that entry. In that case, I would just clear the search in the table and then select the entry?
> 
> What should be done in this case? Should the tab be switched and let it be figured out that the table has been filtered, or should the current search just be cleared?

Current idea: In case the user switches to the tab and the query does not match the entry, the search should be copied

However, this is confusing. Better let the user clear the search for themselves.

**Comment by LoayGhreeb** [comment]
> However, this is confusing. Better let the user clear the search for themselves.

This is what I did in #11010, but after the user clears the search, they have to reselect the entry from the global search window to get it selected in the main table.

**Comment by koppor** [comment]
> > However, this is confusing. Better let the user clear the search for themselves.
> This is what I did in #11010, but after the user clears the search, they have to reselect the entry from the global search window to get it selected in the main table.

Wouldn't be enough if they focus the global search window again? On focus, the global search tries to select the entry again?

**Comment by LoayGhreeb** [comment]
> Wouldn't be enough if they focus the global search window again? On focus, the global search tries to select the entry again?

No, with the current implementation, they will have to deselect the entry and select it again. I don't want to update the selected entry on refocusing because this will lead to automatically switching tabs, which may not be the desired behavior.

**Comment by koppor** [comment]
> > Wouldn't be enough if they focus the global search window again? On focus, the global search tries to select the entry again?
> 
> No, with the current implementation, they will have to deselect the entry and select it again. I don't want to update the selected entry on refocusing because this will lead to automatically switching tabs, which may not be the desired behavior.

Therefore there should be a check: Is the currently selected tab the one of the entry in the global search? - Oh, well, then the global search should know if the last action was a clear search action... Maybe, it can check if there search is empty? If yes, then select the entry? Should (hopefully) be not too disturbing?

**Comment by LoayGhreeb** [comment]
> Therefore there should be a check: Is the currently selected tab the one of the entry in the global search? - Oh, well, then the global search should know if the last action was a clear search action... Maybe, it can check if there search is empty? If yes, then select the entry? Should (hopefully) be not too disturbing?

What about introducing a new property, perhaps in the `StateManager`, to always track the last selected item in the main table? This could also be useful in a similar case with the normal search: if the selected item is not in the search results, and the user then clears the search query, it would lead to no entry being selected.

**Comment by calixtus** [comment]
Identity of the selection. If the entry is selected in the search window, it is selected in the maintable. If it is selected in the maintable, it is selected in the search window. If for whatever reason why (switching the tab, closing the tab whatever) the entry is not selected in the main table, it is not selected in the search window.

**Comment by koppor** [comment]
Maybe, let's keep it simple and do the selection in the main table using a double click in the global search?

We think, IntelliJ does it the same way. <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>F</kbd> opens the global search. On double click, the target is selected and the global search closed. -- Not sure if we should also close the search. Maybe, we can jus drop it into background?

**Comment by LoayGhreeb** [comment]
> Maybe, let's keep it simple and do the selection in the main table using a double click in the global search?

And clear the search query?

> the target is selected and the global search closed. -- Not sure if we should also close the search. Maybe, we can jus drop it into background?

We don't need to create a new `GlobalSearchResultDialog` instance every time the `openGlobalSearchButton` is clicked. Instead, we can reuse the existing instance and just show or hide the window as needed. This will be useful because the query and the selected item in the window will remain the same.

https://github.com/JabRef/jabref/assets/52158423/f5b6203e-01bd-48c2-944d-02974857d358
